### ISSUE TITLE
I-Share uc phrase search with quotes and "Hide e-resources..." returns zero results

### ISSUE THREAD

**Issue by gibsonjc** [CARLI/vufind#417]
Two recent reports/questions about this from SAI and ILC.  Summary of problem seems to be that All Fields, Subject,  and Title searches with a phrase in quotes AND the "Hide e-resources..." filter applied return zero results every time even though not all of the results are e-resources to be hidden. 
Doesn't seem to happen to Author searches, however, as demonstrated in Example 1b below. (All examples below performed in SAI view of I-share uc: https://i-share.carli.illinois.edu/all/vf-sai/)

Example 1a:
In I-Share /all/vf-sai, search for _Kerry James Marshall_ as **All Fields** search

- _without_ quotation marks and _without_ "Hide e-resources..." applied: 158 results
https://i-share.carli.illinois.edu/all/vf-sai/Search/Results?lookfor=Kerry+James+Marshall&type=AllFields&limit=20&sort=relevance

- _with_ quotation marks and _without_ "Hide e-resources..." applied: 106 results
https://i-share.carli.illinois.edu/all/vf-sai/Search/Results?lookfor=%22Kerry+James+Marshall%22&type=AllFields&limit=20&sort=relevance

- _without_ quotation marks and _with_ "Hide e-resources..." applied: 136 results
https://i-share.carli.illinois.edu/all/vf-sai/Search/Results?lookfor=Kerry+James+Marshall&type=AllFields&filter%5B%5D=building%3A%2A

- _with_ quotation marks and _with_ "Hide e-resources..." applied: 0 results
https://i-share.carli.illinois.edu/all/vf-sai/Search/Results?lookfor=%22Kerry+James+Marshall%22&type=AllFields&filter%5B%5D=building%3A%2A

Example 1b:
In I-Share /all/vf-sai, search for _Kerry James Marshall_ as **Author** search

- _without_ quotation marks and _without_ "Hide e-resources..." applied: 34 results
https://i-share.carli.illinois.edu/all/vf-sai/Search/Results?lookfor=Kerry+James+Marshall&type=Author&limit=20&sort=relevance

- _with_ quotation marks and _without_ "Hide e-resources..." applied: 8 results
https://i-share.carli.illinois.edu/all/vf-sai/Search/Results?lookfor=%22Kerry+James+Marshall%22&type=Author&limit=20&sort=relevance

- _without_ quotation marks and _with_ "Hide e-resources..." applied: 33 results
https://i-share.carli.illinois.edu/all/vf-sai/Search/Results?lookfor=Kerry+James+Marshall&type=Author&filter%5B%5D=building%3A%2A

- _with_ quotation marks and _with_ "Hide e-resources..." applied: 8 results (not zero!?!)
https://i-share.carli.illinois.edu/all/vf-sai/Search/Results?lookfor=%22Kerry+James+Marshall%22&type=Author&filter%5B%5D=building%3A%2A

Example 2a:
In I-Share /all/vf-sai, search for _Number Theory_ as **All Fields** search

- _without_ quotation marks and _without_ "Hide e-resources..." applied: 29261 results
https://i-share.carli.illinois.edu/all/vf-sai/Search/Results?lookfor=Number+theory&type=AllFields&limit=20&sort=relevance

- _with_ quotation marks and _without_ "Hide e-resources..." applied: 5181 results
https://i-share.carli.illinois.edu/all/vf-sai/Search/Results?lookfor=%22Number+theory%22&type=AllFields&limit=20&sort=relevance

- _without_ quotation marks and _with_ "Hide e-resources..." applied: 15565 results
https://i-share.carli.illinois.edu/all/vf-sai/Search/Results?lookfor=Number+theory&type=AllFields&filter%5B%5D=building%3A%2A

- _with_ quotation marks and _with_ "Hide e-resources..." applied: 0 results
https://i-share.carli.illinois.edu/all/vf-sai/Search/Results?lookfor=%22Number+theory%22&type=AllFields&filter%5B%5D=building%3A%2A

Example 2b:
In I-Share /all/vf-sai, search for _Number Theory_ as *Subject** search

- _without_ quotation marks and _without_ "Hide e-resources..." applied: 3664 results
https://i-share.carli.illinois.edu/all/vf-sai/Search/Results?lookfor=Number+theory&type=Subject&limit=20&sort=relevance

- _with_ quotation marks and _without_ "Hide e-resources..." applied: 3425 results
https://i-share.carli.illinois.edu/all/vf-sai/Search/Results?lookfor=%22Number+theory%22&type=Subject&limit=20&sort=relevance

- _without_ quotation marks and _with_ "Hide e-resources..." applied: 2638 results
https://i-share.carli.illinois.edu/all/vf-sai/Search/Results?lookfor=Number+theory&type=Subject&filter%5B%5D=building%3A%2A

- _with_ quotation marks and _with_ "Hide e-resources..." applied: 0 results
https://i-share.carli.illinois.edu/all/vf-sai/Search/Results?lookfor=%22Number+theory%22&type=Subject&filter%5B%5D=building%3A%2A

Example 3a:
In I-Share /all/vf-sai, search for _cats and dogs_ as **All Fields** search

- _without_ quotation marks and _without_ "Hide e-resources..." applied: 7023 results
https://i-share.carli.illinois.edu/all/vf-sai/Search/Results?lookfor=cats+and+dogs&type=AllFields&limit=20&sort=relevance

- _with_ quotation marks and _without_ "Hide e-resources..." applied: 311 results
https://i-share.carli.illinois.edu/all/vf-sai/Search/Results?lookfor=%22cats+and+dogs%22&type=AllFields&limit=20&sort=relevance

- _without_ quotation marks and _with_ "Hide e-resources..." applied: 5881 results
https://i-share.carli.illinois.edu/all/vf-sai/Search/Results?lookfor=cats+and+dogs&type=AllFields&filter%5B%5D=building%3A%2A

- _with_ quotation marks and _with_ "Hide e-resources..." applied: 0 results
https://i-share.carli.illinois.edu/all/vf-sai/Search/Results?lookfor=%22cats+and+dogs%22&type=AllFields&filter%5B%5D=building%3A%2A

Example 3b:
In I-Share /all/vf-sai, search for _cats and dogs_ as *Title** search

- _without_ quotation marks and _without_ "Hide e-resources..." applied: 1245 results
https://i-share.carli.illinois.edu/all/vf-sai/Search/Results?lookfor=cats+and+dogs&type=Title&limit=20&sort=relevance

- _with_ quotation marks and _without_ "Hide e-resources..." applied: 64 results
https://i-share.carli.illinois.edu/all/vf-sai/Search/Results?lookfor=%22cats+and+dogs%22&type=Title&limit=20&sort=relevance

- _without_ quotation marks and _with_ "Hide e-resources..." applied: 1019 results
https://i-share.carli.illinois.edu/all/vf-sai/Search/Results?lookfor=cats+and+dogs&type=Title&filter%5B%5D=building%3A%2A

- _with_ quotation marks and _with_ "Hide e-resources..." applied: 0 results
https://i-share.carli.illinois.edu/all/vf-sai/Search/Results?lookfor=%22cats+and+dogs%22&type=Title&filter%5B%5D=building%3A%2A

Use case from SAI:
We usually search the IShare catalog with the 'Hide eresources to which I don't have access' filter on, since we encourage our students to use that so they are not frustrated by getting ebooks that we don't have access too. While teaching students to use this, we also show them how to use quotation marks around an artist's name to help narrow the results.

**Comment by cedelis** [comment]
@gibsonjc , I think I know what's happening here. It has to do with the fact that in searchspecs.yaml, *most* of our queries are using the edismax handler in Solr. The ones that do not use this "enhanced" (the "e" part of "dismax") version are "ExactSettings" definitions (searches that contain quotation marks in the search terms).

Here's an example:

```
Title:
  DismaxFields:
    - title_short^500
    - title_txtP^4500
    - title_full_unstemmed^450
    - title_full^400
    - title^300
    - title_alt^200
    - title_linked^200
    - title_new^100
    - title_old
    - series^100
    - series2
  DismaxHandler: edismax
  ExactSettings:
    DismaxFields:
      - title_full_unstemmed^450
```

Notice above that the edismax is indeed defined (DismaxHandler: edismax). *However*, for the "ExactSettings" portion of the definition, in order to enable the edismax handler, we need to explicitly set it like so:

```
  ExactSettings:
    DismaxFields:
      - title_full_unstemmed^450
    DismaxHandler: edismax
```

So, here's the full definition:

```
Title:
  DismaxFields:
    - title_short^500
    - title_txtP^4500
    - title_full_unstemmed^450
    - title_full^400
    - title^300
    - title_alt^200
    - title_linked^200
    - title_new^100
    - title_old
    - series^100
    - series2
  DismaxHandler: edismax
  ExactSettings:
    DismaxFields:
      - title_full_unstemmed^450
    DismaxHandler: edismax
```

I have a feeling that this was simply an oversight (not realizing the need to explicitly set edismax for ExactSettings searches) as I can't think of any reason why we would have wanted to use dismax for these. In any case, we should probably test various queries to see if we notice any issues with using edismax for these, too. 

I noticed that this has occurred for the following types of searches:

Subject, JournalTitle, Title, AllFields


In TEST, I've made this change, therefore you can run tests vs. Production. If all looks well, I can move this change to prod.

**Comment by gibsonjc** [comment]
@cedelis I have tried several different All Fields searches, both with and without quotes, and with and without "hide e-resources..." checked, and yes, this seems to be behaving more correctly now and providing results instead of zero.  I also tried phrase searches with Title, Subject, and Author, and used other filters along with "hide e-resources...", and got reasonable results.  This is ready for Prod!
